pkg/testutils/klm.go

```diff
@@ -0,0 +1,105 @@
+package testutils
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	apicorev1 "k8s.io/api/core/v1"
+	apimetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kyma-project/lifecycle-manager/pkg/watcher"
+)
+
+const (
+	controlPlaneNamespace = "kcp-system"
+	watcherPodContainer   = "server"
+	KLMPodPrefix          = "klm-controller-manager"
+	KLMPodContainer       = "manager"
+)
+
+var (
+	ErrPodNotFound = errors.New("could not find pod")
+	ErrLogNotFound = errors.New("logMsg was not found in log")
+)
+
+func CheckKLMLogs(ctx context.Context,
+	remoteNamespace string,
+	logMsg string,
+	controlPlaneConfig, runtimeConfig *rest.Config,
+	k8sClient, runtimeClient client.Client,
+	logsSince *apimetav1.Time,
+) error {
+	logs, err := getPodLogs(ctx, controlPlaneConfig, k8sClient, controlPlaneNamespace, KLMPodPrefix, KLMPodContainer,
+		logsSince)
+	if err != nil {
+		return err
+	}
+
+	if strings.Contains(logs, logMsg) {
+		return nil
+	}
+
+	_, err = getPodLogs(ctx, runtimeConfig,
+		runtimeClient, remoteNamespace, watcher.SkrResourceName, watcherPodContainer, logsSince)
+
+	if err != nil {
+		return err
+	}
+	return ErrLogNotFound
+}
+
+func getPodLogs(ctx context.Context,
```

Comment:
Hmm, it makes sense since it's getting bigger, but we need to find a way to group them into packages. I think we need to address this in another issue, along with moving other utilities functions from the tests files themselves.

Reply:
Ok, let's tackle this separately then. Can you create it please?

Reply:
Here is the issue https://github.com/kyma-project/lifecycle-manager/issues/1085